### DISCUSSION TITLE
How can I add a new shop?

### DISCUSSION THREAD

**Discussion by DerHOgi** [jef/streetmerchant#1025] [Help]
What do I have to do, to add a new shop? I tried to add medimax.de to the list of stores.

I created the file:
/src/store/model/medimax.ts with the necessary parameters.

2 files where created by the bot itself:
/build/store/model/medimax.js
/build/store/model/medimax.js.map

I edited the files:
/src/store/model/index.ts
/build/store/model/index.js

I add the store to /.env

But... something is missing.
If I start the bot, I get an error that there is no store named medimax. What is missing?

**Comment by jef** [comment]
Take a look here: https://github.com/jef/streetmerchant/pull/1018

Recently pulled this in. This is how you would add a new store.

Other than that, I'm working on a big update to help make this easier for users.

Thanks!

**Comment by stackenblochen** [comment]
I also wanted to add medimax to the stores list (as well as otto and other smaller stores), how could this be done?

**Comment by DerHOgi** [comment]
> I also wanted to add medimax to the stores list (as well as otto and other smaller stores), how could this be done?

You have to create a new *.ts file for the store e.g. /src/store/model/medimax.ts
Simplest way is to make a copy of an existig store and then ajust all the necessary parameters.

After that, you only have to edit /src/store/model/index.ts and add the new store at two places in the same format like the other stores.

Finally you have to add the store to /.env
That's it.

I have Medimax and Otto up and running but only for sonyps5c.

**Comment by stackenblochen** [comment]
@DerHOgi thanks a lot! would you mind sharing the .ts files, as I don't know the url, container and texts?

**Comment by jef** [comment]
I've recently added this as well: https://github.com/jef/streetmerchant/wiki/Help:-Configuration:-Adding-a-store

Thanks for the help @DerHOgi!

**Comment by w33l** [comment]
> > I also wanted to add medimax to the stores list (as well as otto and other smaller stores), how could this be done?
> 
> You have to create a new *.ts file for the store e.g. /src/store/model/medimax.ts
> Simplest way is to make a copy of an existig store and then ajust all the necessary parameters.
> 
> After that, you only have to edit /src/store/model/index.ts and add the new store at two places in the same format like the other stores.
> 
> Finally you have to add the store to /.env
> That's it.
> 
> I have Medimax and Otto up and running but only for sonyps5c.

@DerHOgi - I'm trying but it is failing stating warn :: No store named smyths, skipping.

Any chance you can help us?

These are the links I found for PS5 i.e.

Argos Console
https://www.argos.co.uk/product/6795199

Argos DE
https://www.argos.co.uk/product/6795151

John Lewis Console
https://www.johnlewis.com/sony-playstation-5-console-with-dualsense-controller/white/p5115192

Smyths Console
https://www.smythstoys.com/uk/en-gb/video-games-and-tablets/playstation-5/playstation-5-consoles/playstation-5-console/p/191259

Smyths DE
https://www.smythstoys.com/uk/en-gb/video-games-and-tablets/playstation-5/playstation-5-consoles/playstation-5-digital-edition-console/p/191430

ShopTo Console
https://www.shopto.net/en/ps5hw01-playstation-5-console-p191472/

ShopTo DE
https://www.shopto.net/en/ps5hw02-playstation-5-digital-console-p195341/

Box Console
https://www.box.co.uk/CFI-1015A-Sony-Playstation-5-Console_3199689.html

Box DE
https://www.box.co.uk/CFI-1015B-Sony-PlayStation-5-Digital-Edition-Conso_3199692.html

Zatu Console
https://www.board-game.co.uk/product/playstation-5/

Zatu DE
https://www.board-game.co.uk/product/playstation-5-digital-edition/

Thanks!

**Comment by holloxob** [comment]
> @DerHOgi thanks a lot! would you mind sharing the .ts files, as I don't know the url, container and texts?

I think the url would just be the page that has the add to cart.

@DerHOgi & @jef  I followed the guide and I am not quite sure how to locate the input for the `container:` in the function of `inStock:`. Many apologies in advance if it is a simple fix, I have little to no programming knowledge.

I am currently trying to create a `.ts` file for called `walmart-ca` for ps5. I copied and paste from `walmart.ts`, but as I am trying to locate and find what I should input for container, I realized the canadian walmart uses very different coding for their site. For example, I am able to locate that the text 'add to cart' in a wrap within a class that has `data-tl-id=ProductPrimaryCTA-cta_add_to_cart_button` when I inspect the element of "add to cart" button, but I cannot locate the same thing on walmart canada. Any suggestion?

**Comment by jef** [comment]
For the containers, what you'll wanna do is use <kbd>F12</kbd> on the site you want to Inspect and click this button

![image](https://user-images.githubusercontent.com/12074633/100685326-2669da80-334a-11eb-93a9-8ac2c659f5f3.png)

Hover over the item you want and it should give you the context: 

![image](https://user-images.githubusercontent.com/12074633/100685310-1e119f80-334a-11eb-91aa-b77b0ff6c2b1.png)

You can also right-click on any website element and select 'Inspect'. That should also give you the same results.

Some people will decide to choose a parent element as it can be unique. Like this case!

I'd rather use `.button spin-button prod-ProductCTA--primary button--primary` instead of `.spin-button-children` as there are probably other elements on the page that are also `.spin-button-children`.

The reason why we use these selectors anyway is to wait for the webpage to load these specific elements, to help eliminate false positives.

**Comment by holloxob** [comment]
Below is my `walmart-ca.ts` file using the very unique selector for the container, but It is still timing out. However, I have tested it by changing the url to US walmart link, and applying your `.button spin-button prod-ProductCTA--primary button--primary` as container for a item that is in stock, it is not timing out but it is also saying that the item is out of stock (I know that is a whole different issue and probably not a good test). Point is I understand what you mean on the container but I am not able to get it to work unfortunately.....

![image](https://user-images.githubusercontent.com/73460193/100709387-de0ae680-3362-11eb-95e1-768aa1ed8200.png)

```
import {Store} from './store';

export const WalmartCa: Store = {
	labels: {
		inStock: {
			container: '.css-1z01by0 e6mlhfu0',
			text: ['Add to cart']
		},
		maxPrice: {
			container: 'span[class*="price-characteristic"]'
		}
	},
	links: [
		{
			brand: 'sony',
			model: 'ps5 console',
			series: 'sonyps5c',
			url: 'https://www.walmart.ca/en/ip/playstation5-console/6000202198562'
		},
		{
			brand: 'sony',
			model: 'ps5 digital',
			series: 'sonyps5de',
			url:'https://www.walmart.ca/en/ip/playstation5-digital-edition/6000202198823'
		},
		
	],
	name: 'walmart-ca'
};
```
